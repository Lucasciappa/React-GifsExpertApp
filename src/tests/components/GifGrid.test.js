import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    const category = "Dragon Ball";
    
    test('debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data : [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={category} /> )
        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imagenes UseFetchGifs', () => {

        const imgs = [{
            id: "asd",
            url:"https://localhost.com",
            title: "Cualquier cosa"
        },
        {
            id: "asd",
            url:"https://localhost.com",
            title: "Cualquier cosa"
        }];

        useFetchGifs.mockReturnValue({
            data : imgs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists() ).toBe(false);
        
        expect(wrapper.find("GifGridItem").length).toBe(imgs.length);

    });

})


