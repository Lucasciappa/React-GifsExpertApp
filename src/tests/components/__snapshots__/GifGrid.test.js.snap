// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Dragon Ball
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes UseFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Dragon Ball
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="asd"
      key="asd"
      title="Cualquier cosa"
      url="https://localhost.com"
    />
    <GifGridItem
      id="asd"
      key="asd"
      title="Cualquier cosa"
      url="https://localhost.com"
    />
  </div>
</Fragment>
`;
